File: 김채린/추가문제/프로그래머스/교점에 별 만들기.java
import java.util.*;

// 검색 풀이 허점이 너무 많았음
class Solution {
    public String[] solution(int[][] line) {
        ArrayList<long[]> arrayList = new ArrayList<>();
        long maxX=Integer.MIN_VALUE;
        long maxY=Integer.MIN_VALUE;
        long minX=Integer.MAX_VALUE;
        long minY=Integer.MAX_VALUE;
        for(int i=0; i<line.length; i++){
            long a= line[i][0];
            long b= line[i][1];
            long e= line[i][2];
            for(int j=i+1; j<line.length; j++){
                long c= line[j][0];
                long d= line[j][1];
                long f= line[j][2];

                long adbc= a*d-b*c;
                if(adbc==0) continue;

                long bfed= b*f-e*d;
                if(bfed%adbc!=0) continue;

                long ecaf= e*c-a*f;
                if(ecaf%adbc!=0) continue;

                long x= bfed/adbc;
                long y= ecaf/adbc;
                arrayList.add(new long[]{x,y});

                maxX = Math.max(x,maxX);
                maxY = Math.max(y,maxY);
                minX = Math.min(x,minX);
                minY = Math.min(y,minY);
            }
        }
        long h = maxY-minY;
        long w = maxX-minX;
        String[] result = new String[(int)h+1];
        String s="";
        for (int i=0; i<=w; i++){
            s+=".";
        }
        Arrays.fill(result,s);
        for (int i=0; i<arrayList.size(); i++){
            long newY = maxY-arrayList.get(i)[1];
            long newX = arrayList.get(i)[0]-minX;
            result[(int) newY]= result[(int) newY].substring(0, (int) newX)+"*"
                +result[(int) newY].substring((int) newX +1);
        }
        return result;
    }
}

File: 이진영/프로그래머스/교점에 별 만들기.js
function solution(lines) {
    const MAX = 100000
    // 교점 구하기
    let set = new Set();
    
    for (let i = 0; i < lines.length; i++) {
        const [a, b, e] = lines[i];
        for (let j = i + 1; j < lines.length; j++) {
            const [c, d, f] = lines[j];
            
            const x = (b * f - e * d) / (a * d - b * c)
            const y = (e * c - a * f) / (a * d - b * c)
            
            if (x % 1 === 0 && y % 1 === 0) {
                set.add(x + " " + y);
            }
        }
    }
    
    // 교점 그리기
    const meets = [...set].map(v => v.split(" ").map(Number));
    const xs = meets.map(v => v[0]);
    const ys = meets.map(v => v[1]);
    
    const [maxX, minX, maxY, minY] = 
          [Math.max(...xs), Math.min(...xs), Math.max(...ys), Math.min(...ys)];
    
    let answer = Array.from(Array(maxY - minY + 1), () => Array(maxX - minX + 1).fill("."));
    
    meets.forEach(([x, y]) => {
        answer[maxY - y][x - minX] = "*";
    })
    return answer.map(v => v.join(""));
}